### ISSUE TITLE
PulseAudio : Segmentation fault (core dumped) 

### ISSUE THREAD

**Issue by moezbenrebah** [pythonarcade/arcade#785]
## Bug Report
A core dumped error raise while playing my game [Space-Survivor](https://github.com/Boul3ez85/Space-Survivor), I'm under ubuntu 20.04 LTS
### Actual behavior:
I used views to handle istructions view, pause view... so when I switch from actual view to the game view in order re-play the view freezed and then it crashed and raise a core dumped error
### Expected behavior:
work fine normally
### Steps to reproduce/example code:
the error msg is below:
```
Exception ignored on calling ctypes callback function: <bound method PulseAudioOperation._success_callback of <pyglet.media.drivers.pulse.interface.PulseAudioOperation object at 0x7fdb408956a0>>
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/pyglet/media/drivers/pulse/interface.py", line 732, in _success_callback
    self._callback()
  File "/usr/local/lib/python3.8/dist-packages/pyglet/media/drivers/pulse/adaptation.py", line 267, in _process_events
    timing_info = self.stream.get_timing_info()
  File "/usr/local/lib/python3.8/dist-packages/pyglet/media/drivers/pulse/interface.py", line 541, in get_timing_info
    timing_info = context.check_ptr_not_null(
  File "/usr/local/lib/python3.8/dist-packages/pyglet/media/drivers/pulse/interface.py", line 379, in check_ptr_not_null
    self.raise_error()
  File "/usr/local/lib/python3.8/dist-packages/pyglet/media/drivers/pulse/interface.py", line 384, in raise_error
    raise PulseAudioException(error, get_ascii_str_or_none(pa.pa_strerror(error)))
pyglet.media.drivers.pulse.interface.PulseAudioException: PulseAudioException: [16] No data
Segmentation fault (core dumped)
```

**Comment by pvcraven** [comment]
This looks to be all pyglet? If so, might need to be reported to them.

**Comment by moezbenrebah** [comment]
yup I think so

**Comment by Cleptomania** [comment]
Is this happening everytime you do something specific? Or just one time and you’re unable to reproduce it?

It seems that it failed to pull the timing info from PulseAudio. My initial thought is this is a one-off glitch in the Matrix type error with PulseAudio. I think it is coming from an issue with the PulseAudio C library at least, not with Pyglet’s implementation as I don’t see where the Python implementation could be causing a segfault like this.

I would recommend seeing if PulseAudio is up to date on your system.

**Comment by moezbenrebah** [comment]
the occurrence of this segmentation fault is random, it happens exactly when I want to return from pause view to game view or from the youwin view to the game view. It's the first time that such an error happens to me since I began to develop my game, also for the PulseAudio, if you want guide me how to update it properly, otherwise I'll search with my own way.

**Comment by einarf** [comment]
Linux users are better off installing OpenAL until this is resolved. The driver is working a lot better.

**Comment by moezbenrebah** [comment]
Yes, I confirm. @einarf with OpenAL sounds effects seems run better.

**Comment by Cleptomania** [comment]
The sound changes have been merged into Arcade 2.5, and seem to be working better now. In 2.5.1 the default load order has been set to prioritize OpenAL and only use other drivers as a fallback if OpenAL is not available.
